### ISSUE TITLE
Binary releases for standalone DCore components

### ISSUE THREAD

**Issue by marian-decent-ch** [DECENTfoundation/DECENT-Network#246]
There are none binary releases for standalone DCore components, nor packages for major Linux systems. It is an issue for deployments it in hosted service and building automated playbooks. Without it, there is a small possibility for accelerated development and integration, or setting up own testnet, or consortium deployments.

**Comment by delatbabel** [comment]
This is from a comment that you made on another ticket:

Can you please try to install decent daemon from DEB package, on ... [Ubuntu LTS 16.04]? Here si the link: https://github.com/marian-decent-ch/DevOps/releases/download/latest_decentd_deb/decentd-0.1.0-Linux-x64-master.deb

You can use also blockchain snapshot from last week:
https://github.com/marian-decent-ch/DevOps/releases/download/bs_180315_1835/blockchain_snapshot_180315_1935.tar.gz

Or you could try Docker image:
docker run -it -p 8090:8090 -p 9090:9090 decentnetwork/decentd decentd

**Comment by delatbabel** [comment]
> Can you please try to install decent daemon from DEB package, on ... [Ubuntu LTS 16.04]? Here si the link: https://github.com/marian-decent-ch/DevOps/releases/download/latest_decentd_deb/decentd-0.1.0-Linux-x64-master.deb

This installs and works fine on a clean Ubuntu 16.04 LTS install but only contains the `decentd` binary which it installs into `/usr/bin` -- it doesn't contain any of the other components such as the CLI wallet, etc.  It would be good to be able to have a binary distribution containing all of the extra tools.  Otherwise compiling from source seems to be the only way forwards.

**Comment by mpmansell** [comment]
I will look into this at my first opportunity.

We are currently overhauling the documentation, with building and distributions being one of the current focus areas.

We are also looking into the possibility of better support for pre-compiled distributions for the major platforms.

**Comment by delatbabel** [comment]
Yes, understood. I did have a read through the testnet documentation on the site mentioned above and it's still not entirely clear, e.g. what config variables can I put into config.ini and which ones must be on the command line, etc.  Also the grammar could use some tidying up.  I will wait for updates.

**Comment by mpmansell** [comment]
I am currently focusing my own efforts on the build and installation processes, but usage/testing and testnet, in particular, will follow straight on from that. I appreciate the input and the ongoing commentary as we want to make sure we have the best documentation possible :)

**Comment by delatbabel** [comment]
Thanks, if there's anything I can help out with then let me know. We're getting some interest from developers here (Vietnam) who would like to have a platform to experiment with so having some working binaries to install and run API queries against is a high priority for me at the moment.

**Comment by jagottsicher** [comment]
I am also very interested in the developement of decentd, cli, and tools "out of the box". I ran the docker file, but it contains "decentd" in version 1.0 only while 1.2 seems to be out already. What was really needed are actual docker files with all stuff already compiled that people can start experiments and start own witness nodes!

I am not happy with compiling, compiling and once again compiling and take care of a working environment! At the moment it feels like there is a tendency among the developers to keep many knowledge under control. 

So is it possible that for the major releases to provide docker files with already  pre-compiled versions of decentd, cli to make it easier for "joe everybody" to start seeding nodes (provide diskspace) and witness-nodes to earn some DCT? That will be a "big leap forward" in the development.

**Comment by mpmansell** [comment]
As it so happens, I am currently working on build documentation that gives as much information, as possible, to people wishing to build from sources. The reality is that developers are not trying to keep things to themselves but, as is usual for anyone familiar with a field, don't necessarily know what others might not know, nor have to the time to stop and address that :) It is this that I am trying to address with upcoming documentation changes.

Regarding pre-built releases, it is our goal to produce as many 'out of the box' solutions as we can, including docker but, as you can understand, in a small team we have to prioritise code integrity and quality and only then can we address multiple builds

**Comment by jagottsicher** [comment]
I already started to build from the source, problem is, that it will not provide a "smart & quick start". Also on VPS with openVZ, low kernel, low mem, no swap you have no chance to build on your own.

but I got your point. while developing you haave to focus on the core, not on dokumentaation, nor on releases, but at least for the milestones aand major releaases, there should be one person (an intern, volunteer) have the job to build a working docker container. people will have to do that 100 times instead of one near the devs does it once only! 

No need for cross compiling for every single OS or linux dist, but a docker container is not that much work to let people fast and easy participate and start their own decent nodes to test it.

**Comment by mpmansell** [comment]
@jagottsicher  May I ask what platform you are building on? I am currently working on a thorough overhaul of the Windows build documentation, and will, once done be moving onto Mac, Linux and general POSIX like systems.

**Comment by marian-decent-ch** [comment]
@jagottsicher Please, see latest builds for Ubuntu LTS, here:
https://github.com/marian-decent-ch/DevOps/releases

Docker images and more current snapshot will be updated within hour or so and more tools to come:

I will provide bash script that will simplify building project from sources, except QT wallet. In addition, I will push builder docker image where most of the libraries are pre-built, so this will be base image for further builds and CI/CD. I am in testing phase. In development is running in kubernetes+mesos and one-click deployments to cloud services. Integrating with cloud APIs is in progress. 

I would appreciate your input.

**Comment by jagottsicher** [comment]
Thanks, I know and I worked worked with ubuntu 16.04, but I and the building docs are okay (well, so lala), but understandable. I was aiming to compile and run a witness node to have a set-it-and-forget-it node running,, which I can afterwards install an several servers. Didn't work out, cause a VPS had too low RAM and it was not possible to install swap.
see #247
So I switched to docker, and @marian-decent-ch helped a lot, but there is no actual version of the docker file. for beginners to become more familiar I think a docker file is a good starting point.

**Comment by marian-decent-ch** [comment]
@jagottsicher I commited pre-release version of very rough bash scripts and Docker files for building Docker images based on Ubuntu Xenial LTS and Decent DEBs:
https://github.com/marian-decent-ch/DevOps/tree/master/Docker

Bash scripts will need more polishing. Later, I will release other images, mostly for building from source code.

Please, let me know your inputs and suggestions.

**Comment by jagottsicher** [comment]
@marian-decent-ch 
today I had only time for y quick check and I also had a look at `https://github.com/marian-decent-ch/DevOps`, which is a big step forward. I am not sure, if you also updated the at [hub.docker.com](https://hub.docker.com/r/decentnetwork/), but from there I got the images.

as said before I had not so much time, but decentd worked fine, but the cli_wallet container works, but seems to be separated from the daemon. I tried to link the cli_wallet container to the ddaemon container, but it seems that it failed.

![wuicktest](https://user-images.githubusercontent.com/30713344/38968548-83872994-43be-11e8-8664-04cdf482d3c4.jpg)

Looks like the network needs to be shared between the conatainers. What do you say?

Anyway: this is a very big step forward to support a growing community of people to run their own witness and seeding nodes worldwide. Go on this way!

**Comment by jagottsicher** [comment]
so I got me a new server with ubuntu 16.04 and installed docker properly. afterwards I started two container in the same network 
Source was [hub.docker.com](https://hub.docker.com/u/decentnetwork/) 

one for the decent daemon daemon i started with 
`docker run -it -p 8090:8090 -p 9090:9090 decentnetwork/decentd:latest decentd`
it seems up and running. And the other one with the cli_wallet I started with
`docker run -it -p 8090:8090 -p 9090:9090 decentnetwork/cli_wallet:latest`

I guess the images at hub.docker.com are up to date. can I ensure that? How to show up the version?


Question are they?

**Comment by marian-decent-ch** [comment]
@jagottsicher 

yes, docker images are up to date. you can check them out via tags: 
https://hub.docker.com/r/decentnetwork/decentd/tags/ and https://hub.docker.com/r/decentnetwork/cli_wallet/tags/

There is also updated wiki on hub.docker.com 
https://hub.docker.com/r/decentnetwork/decentd/ that states commands:

docker run --rm -it --network=host decentnetwork/decentd decentd
docker run --rm -it --network=host decentnetwork/cli_wallet cli_wallet

Also, you can create your own network via "docker network" commands.

**Comment by jagottsicher** [comment]
thanks @marian-decent-ch 
I was at 2018 Global Graphene Developer Conference in Shanghai and now started a bitshares node. That made much things clearer for me, when running Decent.
